### DISCUSSION TITLE
Add `@KeepSorted` annotation?

### DISCUSSION THREAD

**Discussion by magnus-madsen** [flix/flix#7964] [Ideas]
We could add a `@KeepSorted` annotation:

- When placed on an enum, it requires its cases to occur in alphabetical order.
- When placed on a module, it requires the elements to occur in alphabetical order (possibly with blocks of enums, structs, and defs).
- ...

**Comment by jaschdoc** [comment]
Why not just build this into a formatter (if we ever make one)?

**Reply by magnus-madsen** [reply]
It can guide the formatter.
